packages/serverless/src/helper/common.ts

```diff
@@ -50,40 +45,38 @@ export function objectToLookupPattern(
 
 export const gql = (...args: any[]): string => args.join('\n');
 
-export const convertToIndexSpecification = <T = TCollectionIndex>(
-  index: TCollectionIndex
-): TCollectionIndexMap<T> => {
-  const result: TCollectionIndexMap<T> = {};
-  for (const key in index) {
-    if (index[key]) {
-      const fieldKey = `document.${key}.value` as keyof TCollectionIndexMap<T>;
-      if (index[key] === ESorting.Asc) {
-        result[fieldKey] =
-          1 as TCollectionIndexMap<T>[keyof TCollectionIndexMap<T>];
-      } else if (index[key] === ESorting.Desc) {
-        result[fieldKey] =
-          -1 as TCollectionIndexMap<T>[keyof TCollectionIndexMap<T>];
-      }
-    }
-  }
-  return result;
-};
+export const convertIndexToMongoFormat = <T = any>(
+  index: Record<string, EIndexType>
+): TCollectionIndexMap<T> =>
+  Object.entries(index).reduce((acc, [key, value]) => {
+    let val = value === EIndexType.Asc ? 1 : -1;
+    return { ...acc, [`document.${key}.value`]: val };
+  }, {});
+
+export const convertIndexToGraphqlFormat = (
+  index: Record<string, IndexDirection>
+): Record<string, EIndexType> =>
+  Object.entries(index).reduce((acc, [key, value]) => {
+    let val = value ? EIndexType.Asc : EIndexType.Desc;
+    return { ...acc, [key]: val };
+  }, {});
+
 /**
  * Generates an array of indexed field definitions based on a schema.
  *
  * @param schema - An array of schema field definitions, where each field includes
  * properties such as `order`, `index`, `sorting`, and `name`.
  *
- * @returns An array of objects, where each object is of type `Partial<Record<string, ESorting>>`.
- * Each object contains a single field name as the key and its sorting order (`ESorting`)
+ * @returns An array of objects, where each object is of type `Partial<Record<string, EIndexType>>`.
+ * Each object contains a single field name as the key and its sorting order (`EIndexType`)
  * as the value.
  *
  * @example
  * ```typescript
- * const schema: TSchemaFieldDefinition[] = [
```

Comment:
This example is wrong

Reply:
I fixed the example to match with the type `TSchemaFieldDefinition`